### DISCUSSION TITLE
How to use Lingui with SSR?

### DISCUSSION THREAD

**Discussion by wolfpilot** [vercel/next.js#11172] [Help]
Related to #11147 

Given the following requirements:
- Fetch API data before build to enable SSR
- Translate static content based on currently selected language

How can I make Lingui work in this scenario? 

The official example [with-lingui](https://github.com/zeit/next.js/tree/master/examples/with-lingui) works by wrapping a HOC around static routes. However, I need to use `getServerSideProps` which doesn't work with either `getInitialProps` or `getStaticProps`, as noted in #11147 above.

I tried moving the catalog import to `getServerSideProps`, serializing the data (getInitialProps serializes by default), parsing it again and then passing it to a local `I18nProvider`, but the translations stop working because the data's not actually unserialized correctly.

Code sample:

```
import React from 'react'
import { GetServerSideProps } from 'next'
import { I18nProvider } from '@lingui/react'
import { Trans } from '@lingui/macro'

// API
import fetch from 'isomorphic-unfetch'

// Types
import { Catalogs } from '@lingui/core'

type TLanguage = 'en' | 'fr' | 'nl'

interface IProps {
  data: any
  language: TLanguage
  catalogs: Catalogs
}

const TestPage: React.FC<IProps> = ({ data, language, catalogs }) => {
  console.log('data', data)
  console.log('language', language)
  console.log('catalogs', catalogs)

  return (
    <>
      <I18nProvider language={language} catalogs={JSON.parse(catalogs)}>
        <Trans>Some random text</Trans>
        <Trans id="some.random.id" />
      </I18nProvider>
    </>
  )
}

export const getServerSideProps: GetServerSideProps = async ctx => {
  const queryLang =
    ctx.query && typeof ctx.query.lang === 'string' && ctx.query.lang

  const language = queryLang || 'nl'
  const catalog = await import(`../src/locales/${language}/messages.po`).then(
    m => m.default
  )

  const url = 'https://swapi.co/api/people/1'
  const res = await fetch(url)
  const data = await res.json()

  return {
    props: {
      data,
      language,
      catalogs: {
        [language]: JSON.stringify(catalog)
      }
    }
  }
}

export default TestPage

```

I've been going in circles for hours and can't really find any examples of how it's supposed work. Any help would be appreciated, thank you!

**Comment by sodhisaab** [comment]
did you find any solution? 🤔
